scripts/extend_index_yaml_test.py

```diff
@@ -30,8 +30,10 @@ class ExtendIndexYamlTest(test_utils.GenericTestBase):
 
     def setUp(self) -> None:
         super().setUp()
-        self.index_yaml_file = tempfile.NamedTemporaryFile()
-        self.web_inf_index_yaml_file = tempfile.NamedTemporaryFile()
+        with tempfile.NamedTemporaryFile() as index_yaml_file:
+            self.index_yaml_file = index_yaml_file
+        with tempfile.NamedTemporaryFile() as web_inf_index_yaml_file:
+            self.web_inf_index_yaml_file = web_inf_index_yaml_file
```

Comment:
As these two are only used in `_run_test_for_extend_index_yaml` (through `self.index_yaml_file_name` and `self.index_yaml_swap`) you can just move them there. That should help with the issue you're facing.

Reply:
Do you mean that I should move all code from `setup()` to `_run_test_for_extend_index_yaml()`? Because `index_yaml_file` and `web_inf_index_yaml_file` are used in many places in the `setup()` function. @vojtechjelinek

Reply:
Not sure if all, but if the variables that depend on `index_yaml_file` or `web_inf_index_yaml_file` probably yes. You will have to try it out.

Reply:
Hi @vojtechjelinek, when changing as you suggested the linter checks fail with error:

```
************* Module scripts.extend_index_yaml_test
scripts/extend_index_yaml_test.py:31:4: W0235: Useless super delegation in method 'setUp' 
...
scripts/extend_index_yaml_test.py:38:12: W0201: Attribute 'index_yaml_file' defined outside __init__ 
scripts/extend_index_yaml_test.py:40:12: W0201: Attribute 'web_inf_index_yaml_file' defined outside __init__ 
```

Reply:
@richard-johansson I do not now the exact code that you have right now. Please send it too. Also, did you try to search for these problems online and solve them?